### PR TITLE
nearly pure re-factor: split out code into its own modules.

### PR DESCRIPTION
**Pull request by mmeeks**
StateRecorder.hpp split from ChildSession.cpp
KitWebSocketHandler.[ch]pp split from Kit.cpp.
ThreadPool.hpp split from RenderTiles.hpp

Expose headers for KitSocketPoll and Document
at the same time.

Not clear we need the DocumentManagerInterface anymore.

Conditionally compile out Document::createSession for unittest dependency breaking, and avoid Rlimit::handleSetrlimitCommand likewise.

Make makePropertyValue a private method of Kit.cpp.

clang-format new files.

Change-Id: I47a1d6afe20165f156b477a931b94c916cff4b9d

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Review by Ashod — APPROVED** [review]
Thanks, @mmeeks. Looks good.

File: ios/Mobile.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -67,6 +67,7 @@
 		BE8D85D6214055F3009F1860 /* rc in Resources */ = {isa = PBXBuildFile; fileRef = BE8D85C8214055F3009F1860 /* rc */; };
 		BE9ADE3F265D046600BC034A /* TraceEvent.cpp in Sources */ = {isa = PBXBuildFile; fileRef = BE9ADE3D265D046600BC034A /* TraceEvent.cpp */; };
 		BEA2835621467FDD00848631 /* Kit.cpp in Sources */ = {isa = PBXBuildFile; fileRef = BEA2835521467FDD00848631 /* Kit.cpp */; };
+		BEA2835621467F0D00848631 /* KitWebSocket.cpp in Sources */ = {isa = PBXBuildFile; fileRef = BEA2835521467F0D00848631 /* KitWebSocket.cpp */; };
```
Comment (Ashod) [inline ios/Mobile.xcodeproj/project.pbxproj:70]: I manually change the IDs to make them unique. I assume being unique is important. But perhaps this works too.

Comment (mmeeks) [reply ios/Mobile.xcodeproj/project.pbxproj:70]: You'll notice that FDD and F0D are different in the middle =)

**Comment by mmeeks** [comment]
Thanks for merging.

**Pull request merged**